### ISSUE TITLE
Show only one clipboard element instead of the list

### ISSUE THREAD

**Issue by zsbodnar** [ffMathy/Shapeshifter#54]
Sometimes two lines of preview is not enough, it would be good to see only one clipboard item at a time (but in a bigger window) instead of the list. Just like in Flycut for Mac (http://a4.mzstatic.com/us/r30/Purple/v4/0d/aa/55/0daa5524-f58b-a5bf-be16-ed4219c67318/screen800x500.jpeg)
Ideally it would be good to make it configurable on the settings page (list vs. single item) together with the preview window size, position and transparency.

**Comment by ffMathy** [comment]
Another great idea by you! We definitely need this. Good job. I think the Flycut screenshot is ugly though. We can do it better.

**Comment by ffMathy** [comment]
We could implement it as a kind of "big picture" mode where the UI is centered and the active item in the list is way bigger and more detailed.

**Comment by thegabriestein** [comment]
@zsbodnar, this could be really valuable! Should the title be: "Increased content previews for larger clippings", like when you have a whole paragraph and want to see more of a preview of it, or do you mean that it should only show a single item at a time as a preference?

**Comment by thegabriestein** [comment]
@ffMathy Haven't heard back on this yet, but potentially worth closing since this is technically already available through the "Maximum number of items in the clipboard deck" could just be set to "1". 

Maybe add a setting for "Number of lines of preview" for each item? It wasn't usually a huge concern in v5 with how much it shows, and v6 is about the same. But it could make sense if you're copying and pasting large blocks of code or text or something. I don't feel like it's a huge concern though.

In the future, potentially as  you scroll down the list of items it would expand the item you're hovering on, kind of like the Mac dock magnification? This would take up less space for items that aren't selected, but show more of the currently selected item. 
![](https://i.imgur.com/oOU2L7K.png)

**Comment by ffMathy** [comment]
I don't think that's what is proposed. Imagine a UI without a list at all. In the center of your screen in a semi transparent way you just see one item, but you can still copy more.

As you use the arrows to navigate through, only the selected item is shown. 

The way this item is shown is a very big representation that looks entirely different than the current list. For an image, it could be displaying the image itself. For text, it could be some of the text shown etc.

**Comment by zsbodnar** [comment]
Hi @ffMathy, thanks for clarification, I was suggesting exactly that, just one big semi-transparent popup and no list. The list view is useful if you use the mouse so it is possible to quickly select a "random" item. But developers mostly use the keyboard so we have to navigate through all the elements anyway.

**Comment by thegabriestein** [comment]
So the issue is how the items are previewed? And that some items like pictures, code snippets, paragraphs etc need more preview (and properly formatted preview) to determine if they're correct before pasting? 

I would posit that the primary benefit of Shapeshifter is a historical list of items that can quickly be scanned and selected using keyboard and then redeployed quickly. But yes, if those items can't be previewed fully, that would be limiting.

Out of 19 items currently on my Shapeshifter at the time of writing, all of them fit on a single preview line. But I do think a larger preview option is a good idea for certain items like pictures or paragraphs of text to make sure you're about to paste the right item version (as mentioned a little in #245). 

Three solutions come to mind:
1. What about adding an action menu item for "Preview" which when selected through the right arrow key expands the item preview, and pressing the left arrow key collapses the preview?
2. The swell magnification type option as I mentioned above, and potentially this would be great to add either way and automatically trigger for any item larger than the 2-3 preview lines?
3. Adding a settings menu item for "Large preview mode" where you use keyboard to see a single item at a time

However, for the average use case, being able to immediately see at least 3 recent items is very useful since it's very common over a session to go back in the history more than 3+ items. My average is probably 5-10 items back (for commands, snippets, etc). Much easier to visually scan a list before choosing an item than have to flip through them all manually.

**Comment by ffMathy** [comment]
While I still buy the idea of having the lost as default because it covers most cases, I also like the idea of a big picture mode because I think many will rarely go back more than 2 or 3 items at a time.

Don't worry, not going to kill the list.

**Comment by zsbodnar** [comment]
Yes, my feature request is about to have a bigger preview mode mostly for bigger code snippets, etc. 
I agree that it's easier to visually scan a list before choosing an item but you have to flip through them manually anyway when you're using the keyboard (making the list with 1-2 line height items kinda unnecessary, at least to me).
I vote for option nr. 3.

**Comment by ffMathy** [comment]
Option 3 it is then 👌
